wazuh/wazuh-kibana-app
# [Enhancement] Consistency in the non-core visualizations and agent status

# Description

This PR adds some enhancements in the consistency of non-core visualizations and agent status.

# Changes
- Consistency in custom donut visualizations:
 - Replaced the visualizations in `Agents` and `Modules/SCA/Inventory` by the `VisualizationBasic` component
 - Removed the `Pie` component
- Consistency in agent status, labels, and colors:
 - Centralized the agent status, labels, and colors to constants.ts file
 - Replaced the hardcoded values with the centralized ones
 - Added the `pending` status to:
   - Agents stats in Modules directory
   - Status visualization in `Agents` section
   - Details panel in `Agents` section
   - Agents stats in Management/Status
- Created a service to get the color and label depending on the agent
  status. It is shared by the backend and frontend.
- Created a component to display the agent status: AgentStatus.
  It is adapted to the different use cases.
  - Replaced the old rendering with the new one in some components. Removed unnecessary methods and layout.
- Removed deprecated methods and files
  - Rendering agent status stats when the monitoring job was disabled
    and the visualization of agent status evolution was in Modules/Security events. `WzVisualize` component.
    - Removed an API request to get the agent status data
- Refactored the API request managing in:
  - `Modules/SCA/Inventory`
  - `Agents`
  - `Management/Status`
- Added a new property `error` to `VisualizationBasic` to render the
  error message coming from `error.message`
- Fixed an error when creating a visualization that didn't apply the
  definitions in `uiStateJSON` property. This is due to a bug in Kibana 7.10.2
- Fixed some typos

# Screenshots
- Agents
![image](https://user-images.githubusercontent.com/34042064/170266950-4320d026-bc16-407e-9abe-a17c3448dc04.png)
- Modules/SCA
![image](https://user-images.githubusercontent.com/34042064/170265879-23c25db1-c45b-4af7-aae2-43fde1bccc79.png)
- Modules directory
![image](https://user-images.githubusercontent.com/34042064/170266224-523a2c02-26a0-48bb-a4fe-75fab4ff9691.png)
- Management/Status
![image](https://user-images.githubusercontent.com/34042064/170266066-87f763e5-bbe6-4137-b55e-be0d4b179115.png)
- Agent selector
![image](https://user-images.githubusercontent.com/34042064/170267104-90ab1eb7-8eba-48ef-9ddc-12b3427fb2ae.png)
- Group's agent
![image](https://user-images.githubusercontent.com/34042064/170267254-9bb8b05a-c209-4dcf-9c42-60539a522617.png)

# Test
- Agents:
  - Status visualization:
    - Click on the legend's labels adding a filter to the search bar
    - Include the `Pending` status
  - Details panels:
    - Click on the agent status stat adding a filter to the search bar
    - Include the `Pending` status
  - Evolution visualization:
    - status colors are equals to the other panels
  - Search bar
    - `status` suggestion includes `pending` status
- Agent selector
  - `status` suggestion includes `pending` status
  - `Status` colum displays the status indicator and the label.
  - If there is a selected agent, it displays the status indicator and the agent name (id) as a label.
- Management/Groups/<Group>
  - Search bar:
    - `status` suggestion includes `pending` status
- Modules directory
  - Display the agent count by status including `pending` state
- Management/Status
  - Display the agent count by status including `pending` state
  - Display the agent status using the labels.
  - The agent count should display as expected when loading the first time or change the manager using the manager node selector for a cluster mode
- Reviewing the files changed, access to the sections, and using an agent with status same of change, see that is working as expected.
  - `Modules/File integrity monitoring/Inventory`
    - `active`: display the data
    - not `active`: display a prompt related to the agent status
  - `Modules/SCA/Inventory`
    - `never_connected`: prompt with a message related to the agent status
    - not `never_connected`: display the data
  - `Agent/Stats`:
    - `active`: display the data
    - not `active`: display a prompt related to the agent status
  - `Agent/Inventory`
    - `active`: display the data
    - `disconnected`: display a prompt related to the agent status
    - `never_connected`: display a prompt related to the agent status
  - `Agent/Overview`:
    - `never_connected`: display a prompt related to the agent status
    - no `never_connected`: display the data
  - `Agents`
    - Table
      - not `never_connected`: not display the button to open the agent configuration
      - `Status` column is rendered with an indicator that depends on the agent status and a label without `_` character
        - For window width < 1200, only display the indicator and no the status label
  - `Agent/Configuration`
    - `active`: display the section
    - `never_connected`: display a prompt related to the agent status
  - `<Module> - Generate a report in PDF`
    - not `active`: display a message with the current state of the agent.
- The order of agent status are: `active`, `disconnected`, `pending`, `never_connected` 
    

Closes #4077
